### DISCUSSION TITLE
How to run all tests using commandline without closing runner after each spech

### DISCUSSION THREAD

**Discussion by Ahmar828** [cypress-io/cypress#16870] [Questions and Help]
Hi,

I wrote hooks in support/index.js file. In after() hook i am inserting reports data into DataBase. 

![Screenshot from 2021-06-09 13-09-13](https://user-images.githubusercontent.com/78353735/121318885-1a785a00-c925-11eb-8370-d7b727b0dc85.png)

Now issue is, When I run all of my spec files through interface it is working fine because it is running all the spec files together and data from after() hook is inserting into data base.

![Screenshot from 2021-06-09 13-10-01](https://user-images.githubusercontent.com/78353735/121319229-6fb46b80-c925-11eb-9739-17c0b91149f7.png)

![Screenshot from 2021-06-09 13-10-37](https://user-images.githubusercontent.com/78353735/121319291-7e9b1e00-c925-11eb-8b88-89c6411a405a.png)

But when I run from command line using script `Cypress run` it runs all the spec files but one by one. It opens 1st spec file run all test cases and closes spec file. when it opens 2nd spec it reinitialize all the variables in support/index.js due to which data from last spec got lost.

ideally on running with `Cypress run` it should run all spec files together. 

please suggest solution.

**Comment by jennifer-shehane** [comment]
This is the difference between `cypress run` and `cypress open` explained here: https://github.com/cypress-io/cypress/issues/1586

You could do something like run `cypress run --spec spec1.js`, `cypress run --spec spec2.js` with each spec separately and have some scripts that run after each one, depending on what you want to achieve.

**Reply by Ahmar828** [reply]
@jennifer-shehane Thanks for your response, But I am unable to achieve the behavior I required. In Simple words i want to run all spec files without closing browser in between spec files using `cypress run` even if I use `cypress run --spec spec1.js,spec2.js` it runs spec1.js close the browser, reopens the browser to execute spec2.js due to which hooks in support/index.js are working individually for each file.

ideally on running these two specs flow should be like 
before()
beforeEach()
spec1.js / test1
afterEach()
beforeEach()
spec2.js /test1
afterEach()
after()

But it is working as

before()
beforeEach()
spec1.js / test1
afterEach()
after()
before()
beforeEach()
spec2.js 
afterEach()
after()

**Reply by jennifer-shehane** [reply]
This is not possible. The browser shuts down in between each spec file that is run. There's a request for this behavior to change here, but this is not on our current roadmap: https://github.com/cypress-io/cypress/issues/2951
